### ISSUE TITLE
Window can be moved when maximized or fullscreen

### ISSUE THREAD

**Issue by ilovelinux** [awesomeWM/awesome#2265]
Output of `awesome --version`:

awesome v4.2 (Human after all)
 • Compiled against Lua 5.3.4 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.5
 • LGI version: 0.9.2

**How to reproduce the issue:**

- Install Firefox (Base or Nightly)
- Open it and maximize the window (`mod+m`)
- Click and hold the empty space all around the address bar
- Move the mouse

**Actual result:**

You can move Firefox's window even if it's maximized (#2135?)

**Expected result:**

It shouldn't

**Comment by actionless** [comment]
that intended to be fixed in this https://github.com/awesomeWM/awesome/issues/2036

cc @blueyed

**Comment by ilovelinux** [comment]
I can't move the window when it's fullscreen

**Comment by actionless** [comment]
mb because i have slightly different keybinding:
https://github.com/actionless/awesome_config/blob/devel/config/keys.lua#L740-L743
https://github.com/actionless/awesome_config/blob/devel/config/keys.lua#L624-L643

**Comment by ilovelinux** [comment]
Where do I find these files? I'm on Archlinux

**Comment by actionless** [comment]
those files are extension to my rc.lua

**Comment by actionless** [comment]
with default rc.lua i guess you still should have such behavior if will be on floating layout, maximize/fullscreen a client and next will move it with the keyboard

**Comment by alfunx** [comment]
I can't reproduce this on the Git master version, using the default config and Firefox (as mentioned by OP).

**Comment by actionless** [comment]
also, i've checked, with mouse it also reproducing for me, not only with the keyboard

**Comment by alfunx** [comment]
> with mouse it reproducing for me as well

Okay, that's weird. For me it seems like it's been fixed in `678ead26` (like you mentioned). Before that, I can reproduce this issue, with the mouse as well as the keyboard. Not sure if I misunderstood you...

**Comment by ilovelinux** [comment]
Just compiled AwesomeWM from sources and I can confirm that the issue has been fixed! Good work!
Someone should update the package on the archlinux's repos! It is almost 1 year old
```
$ pacman -Qi awesome
Name            : awesome
Version         : 4.2-1
Description     : Highly configurable framework window manager
Architecture    : x86_64
URL             : https://awesomewm.org/
Licenses        : GPL2
Groups          : None
Provides        : notification-daemon
Depends On      : cairo  dbus  gdk-pixbuf2  imlib2  libxdg-basedir  lua
                  lua-lgi  pango  startup-notification  xcb-util-cursor
                  xcb-util-keysyms  xcb-util-wm  xorg-xmessage
                  libxkbcommon-x11  libxkbcommon  xcb-util-xrm
Optional Deps   : rlwrap: readline support for awesome-client
                  dex: autostart your desktop files
                  vicious: widgets for the Awesome window manager
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 8.33 MiB
Packager        : Levente Polyak <anthraxx@archlinux.org>
Build Date      : Wed 19 Jul 2017 12:25:41 PM CEST
Install Date    : Thu 12 Apr 2018 07:01:21 PM CEST
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Comment by actionless** [comment]
i indeed can't reproduce it with the default config btw, but i am still not sure it's fine, for example i can reproduce it with default config if will add those lines and try resize a maximized/fullscreen client:
```lua
    awful.button({ modkey, "Control" }, 1, function(c)
      awful.mouse.resize(c, nil, {jump_to_corner=false})
    end),
```

**Comment by mvduin** [comment]
> > with mouse it reproducing for me as well
> 
> Okay, that's weird. For me it seems like it's been fixed in 678ead26 (like you mentioned). Before that, I can reproduce this issue, with the mouse as well as the keyboard. Not sure if I misunderstood you...

That commit blocks requests from the client itself, typically used when the client (e.g. Firefox) itself provides user interface elements to move or resize the window (rather than letting the WM deal with that). The original post's example clearly falls in this category, hence was fixed by 678ead26.

That commit however has no effect on moving or resizing a window using `awful.mouse.client.move`/`resize`, which the default config calls when clicking in the titlebar of a window or modkey+clicking anywhere in the window. These functions still do not honor immobilized_{horizontal,vertical} today, and consequently you can move/resize a window that is maximized vertically or horizontally (tested in awesome 4.3 in default config).

They do ignore the click entirely if certain conditions are met and 51ddb563 added "window is maximized" to that list of conditions.
